docs/overview.md

```diff
@@ -0,0 +1,92 @@
+
+Most of the time, system administrators use the Content app as part of the bigger image building workflow - when they need to create an image with particular version of repositories in it. Those images they then use to spin up servers in bulk.
+
+The future goal is to allow users to define only templates as a use-case, removing the Content app action of adding a repository from the navigation bar. Adding a repository action will then be a part of a template definition. More on Content app actions follows below.
+
+### What can users do with the Content app?
+
+The Content app allows users to choose and define which versions of [packages](https://en.wikipedia.org/wiki/Package_format) contained in a **repository** (collection of packages) they want to keep track of. So Content app is a bit like the GIT version control system, only for .rpm packages and repositories. Plus there runs a regular service that checks for updates of repositories and saves any new versions into a database. A repository can be coming from Red Hat or from a third-party (a custom repository).
+
+Users can also define which repositories they want to use in a **template**. Template works like a blueprint that puts 2 and more repositories together.
+
+So there are 3 layers - packages(1.), which are contained in repositories(2.), which can be mixed into templates(3.).
+
+Content app currently works only with the YUM repository format type.
+
+The versioning capability of the Content app is backed by the Pulp Server Service, which is maintained by the Pulp team.
+
+[Learn more about the architecture and dependencies](./architecture.md)
+
+### Content app Capabilities / Actions
+
+1. Adding a Repository
+
+   There are 3 ways to add a repository:
+
+   A: introspect only  
+   First you need to provide a URL containing a repository. This option only introspects the provided URL and creates metadata about the repo, which are then saved into a DB. Plus there is a service that continuously checks for updates of the tracked repositories.
+
+   [Read more about introspection](./workflows/introspection.md)
+
+   B: snapshot  
+   This is the default option. It introspects the provided url and creates metadata, plus it copies the whole repository into a DB. There is also a service that continuously keeps checking the repo for any changes and creates history of snapshotted versions for the repo. These snapshots can then be used to build an image.
+
+   Snapshot making depends on the [tasking system](./tasking_system/tasking_system.md).
+
+   [Read more about snapshotting](./workflows/snapshotting.md)
+
+   C: upload  
+   This option behaves like the snapshot one but working on private/secret repositories. So instead of providing a public URL, users upload a repository, which is then saved as a snaphot into a DB.
```

Comment:
```suggestion
   This option behaves like the snapshot one but for local private repositories. So instead of providing a public URL, users upload a repository, which is then saved as a snaphot into a DB.
```